addon/mixins/select.js

```diff
@@ -0,0 +1,44 @@
+import Mixin from "@ember/object/mixin";
+
+export default Mixin.create({
+  setLabel(value) {
+    if (typeof value.label === "string") {
```

Comment:
Maybe a `switch/case` instead?

Reply:
so...i guess with a switch/case you can't check `typeof` as a case, ie.

``` switch (value) {
      case value.label === "string":
        this.set("label", value.label);
        break;
      case (typeof value === "string" || typeof value === "number"):
        this.set("label", value);
        break;
```

to make it work i would need to do either.

1.  this would not work for all my cases
``` switch (typeof value) {
      case  "string":
        this.set("label", value);
        break;
      case "number":
        this.set("label", value);
        break;
```

or 

2.
``` switch (value) {
      case value.label === "string":
        this.set("label", value.label);
        break;
      case value: 
        if (typeof value === "string" || typeof value === "number") {
          this.set("label", value);
       }
        break;
```

This last option seems just as ugly as the current `if/else if` statement, so I think I am leaning towards keeping it as is.  Let me know if you object or if this suggestion was more of a 🍹

Reply:
Yeah, every language has different limitations on their switch/case like construct, and I wasn't sure of JS's. Leave it as-is.